docs/topics/Threads.md

```diff
@@ -0,0 +1,111 @@
+
+Because of this constraint, the gateway protocol is designed to ensure that bots are able to have an accurate view of the full set of active threads, but archived threads are not synced up-front via the gateway.
+
+Threads do not count against the max-channels limit in a guild, but there will be a new limit on the maximum number of active threads in a guild.
+
+## Permissions
+
+Threads generally inherit permissions from the parent channel. If you can send messages or add reactions in the parent channel, you can do that in a thread as well.
+
+Two new permission bits will be added, `USE_PUBLIC_THREADS` and `USE_PRIVATE_THREADS`. Users can _create_ a thread if they have BOTH the `SEND_MESSAGES` permission and the appropriate `USE_THREADS` permission. Users can _send messages_ in a thread if they have EITHER the `SEND_MESSAGES` permission or the appropriate `USE_THREADS` permission.
+
+Private threads are similar to Group DMs, but in a guild: You must be invited to the thread to be able to view or participate in it, or be a moderator (`MANAGE_THREADS` permission).
+
+Finally, threads are treated slightly differently from channels in the gateway protocol. Clients will not be informed of a thread through the gateway if they do not have permission to view that thread.
+
+## Gateway Events
+
+- [Guild Create](#DOCS_TOPICS_GATEWAY/guild-create) contains a new field, `threads`, which is an array of channel objects. This represents all active threads in the guild, that the current user is able to view.
+- When a thread is created, updated, or deleted, a [Thread Create](#DOCS_TOPICS_GATEWAY/thread-create), [Thread Update](#DOCS_TOPICS_GATEWAY/thread-update), or [Thread Delete](#DOCS_TOPICS_GATEWAY/thread-delete) event is sent. Like their channel counterparts, these just contain a thread.
+- Since the gateway only syncs active threads that the user can see, if a user _gains_ access to a channel, then the gateway may need to sync the active threads in that channel to the user. It will send a [Thread List Sync](#DOCS_TOPICS_GATEWAY/thread-list-sync) event for this.
+
+## Thread Membership
+
+Each thread tracks explicit membership. There are two primary use cases for this data:
+
+- Clients use _their own_ [thread member](#DOCS_RESOURCES_CHANNEL/thread-member-object) to calculate read states and notification settings. This is largely irrelevant for bots though, but is the reason for some of the syncing complexity detailed here.
+- Knowing everyone that is in a thread.
+
+Membership is tracked in an array of [thread member](#DOCS_RESOURCES_CHANNEL/thread-member-object) objects. These have four fields, `id` (the thread id), `user_id`, `join_timestamp`, and `flags`. Currently the only `flags` are for notification settings, but others may be added in future updates.
```

Comment:
If a `thread member` loses "access" by way of the removal of `SEND_MESSAGES`, or `USE_PUBLIC/PRIVATE_THREADS` are they removed from this array?

If not, are they removed if `READ_MESSAGES` is removed?

Reply:
No, we don't remove people when losing access to a channel.
One reason is a moderator may make a mistake or temporarily want to hide a channel, and it would be a bad experience to kick everyone out of a thread then!